pkg/client/generic/genericclient.go

```diff
@@ -43,7 +43,8 @@ type genericClient struct {
 }
 
 func New(config *rest.Config) (Client, error) {
-	scheme := apis.Scheme
+	scheme := runtime.NewScheme()
+	apis.AddToScheme(scheme)
```

Comment:
(no action required) we could alternatively do call AddToScheme of k8s and cluster-registry in init() of federation api package (like the sigs.k8s.io/controller-runtime does it). Calling `runtime.NewScheme` should also be fine as it is called once per new client.

Reply:
Given that the kube client appears to use a global, I think that would be a preferable strategy.

Reply:
does fed v2 types depend on clusterregistry types? otherwise, it seems a bit awkward to me to AddToScheme there.

@shashidharatd where does controller-runtime do it? i couldn't find it by a simple grep.

Reply:
@yamt, please fix this. Thanks!

Reply:
@yamt, sorry, i meant, in the new kubebuilder(>=v1.0.5). you can find that when a new type is created. there is also an auto generated file to add the Scheme to the global variable. like below
```
func init() {
	// Register the types with the Scheme so the components can map objects to GroupVersionKinds and back
	AddToSchemes = append(AddToSchemes, v1alpha1.SchemeBuilder.AddToScheme)
}
```

Reply:
i pushed a new patch after a private chat with @shashidharatd